File: HeartForCharity/UI/heartforcharity_desktop/lib/model/responses/donation.dart
class Donation {
  final int donationId;
  final int campaignId;
  final String? campaignTitle;
  final int? userProfileId;
  final String? donorName;
  final double amount;
  final bool isAnonymous;
  final String? payPalTransactionId;
  final String? status;
  final DateTime? donationDateTime;

  Donation({
    this.donationId = 0,
    this.campaignId = 0,
    this.campaignTitle,
    this.userProfileId,
    this.donorName,
    this.amount = 0,
    this.isAnonymous = false,
    this.payPalTransactionId,
    this.status,
    this.donationDateTime,
  });

  factory Donation.fromJson(Map<String, dynamic> json) => Donation(
        donationId: json['donationId'] ?? 0,
        campaignId: json['campaignId'] ?? 0,
        campaignTitle: json['campaignTitle'],
        userProfileId: json['userProfileId'],
        donorName: json['donorName'],
        amount: (json['amount'] as num?)?.toDouble() ?? 0,
        isAnonymous: json['isAnonymous'] ?? false,
        payPalTransactionId: json['payPalTransactionId'],
        status: json['status'],
        donationDateTime: json['donationDateTime'] != null
            ? DateTime.parse(json['donationDateTime'])
            : null,
      );

  Map<String, dynamic> toJson() => {
        'donationId': donationId,
        'campaignId': campaignId,
        'campaignTitle': campaignTitle,
        'userProfileId': userProfileId,
        'donorName': donorName,
        'amount': amount,
        'isAnonymous': isAnonymous,
        'payPalTransactionId': payPalTransactionId,
        'status': status,
        'donationDateTime': donationDateTime?.toIso8601String(),
      };
}

File: HeartForCharity/HeartForCharity.Services/Database/Donation.cs
using HeartForCharity.Model.Enums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HeartForCharity.Services.Database
{
    public class Donation
    {
        [Key]
        public int DonationId { get; set; }

        [Required]
        [ForeignKey(nameof(Campaign))]
        public int CampaignId { get; set; }

        [ForeignKey(nameof(UserProfile))]
        public int? UserProfileId { get; set; }  

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        public bool IsAnonymous { get; set; } = false;

        [MaxLength(255)]
        public string? PayPalOrderId { get; set; }

        [MaxLength(255)]
        public string? PayPalTransactionId { get; set; }

        public DonationStatus Status { get; set; } = DonationStatus.Pending;

        public DateTime DonationDateTime { get; set; } = DateTime.UtcNow;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public virtual Campaign Campaign { get; set; } = null!;
        public virtual UserProfile? UserProfile { get; set; }
    }
}